### PR TITLE
Updated README for iOS installation instructions

### PR DESCRIPTION
**Pull request by dcaunt**
Implements #480 

It seems that in Xcode 8.3, at least, adding the binary to the "Embedded Frameworks" build phase does all that's required. Runs fine on device.

We don't mention Carthage as an installation option, but we have the badge. Think it's worth a mention above CocoaPods?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -480,12 +480,11 @@ To add Mantle to your application:
...
-    "Link Binary With Libraries" phase.
-    * **On iOS**, add `libMantle.a`.
-    * **On OS X**, add `Mantle.framework`. Mantle must also be added to any
-      "Copy Frameworks" build phase. If you don't already have one, simply add a
-      "Copy Files" build phase and target the "Frameworks" destination.
+ 1. On the "General" tab of your application target, link Mantle.
+    * **On iOS**, add `Mantle.framework` to the "Embedded Binaries" build phase.
+    * **On OS X**, add `Mantle.framework` to the "Linked Frameworks and Libraries" build phase. 
+    Mantle must also be added to any "Copy Frameworks" build phase. If you don't already have 
+    one, simply add a "Copy Files" build phase and target the "Frameworks" destination.
```
Comment (jspahrsummers) [inline README.md]: There should be an Embedded Binaries section on OS X too.

File: README.md [hunk 2]
```diff
@@ -480,12 +480,11 @@ To add Mantle to your application:
...
  1. Drag and drop `Mantle.xcodeproj` into your application's Xcode project or
     workspace.
- 1. On the "Build Phases" tab of your application target, add Mantle to the
-    "Link Binary With Libraries" phase.
-    * **On iOS**, add `libMantle.a`.
-    * **On OS X**, add `Mantle.framework`. Mantle must also be added to any
-      "Copy Frameworks" build phase. If you don't already have one, simply add a
-      "Copy Files" build phase and target the "Frameworks" destination.
+ 1. On the "General" tab of your application target, link Mantle.
+    * **On iOS**, add `Mantle.framework` to the "Embedded Binaries" build phase.
```
Comment (jspahrsummers) [inline README.md]: It's not a build phase, but just a pane on the General tab. Can you rephrase that slightly?

**Comment by jspahrsummers** [comment]
Thanks @dcaunt!

> We don't mention Carthage as an installation option, but we have the badge. Think it's worth a mention above CocoaPods?

Yep, I think that'd be great. :sparkles:

File: README.md (hunk 1 above)
Comment (dcaunt) [inline README.md]: This simplifies things, now both platforms have the same steps

**Comment by archagon** [comment]
FYI, I think there's a slight correction to be made. If you add the Mantle xcodeproj to your workspace, you won't be able to drag/drop the Mantle.framework to the embedded binaries list. You can only do that if you add the Mantle xcodeproj to the main project xcodeproj inside the workspace, unless I'm missing something.

File: README.md [hunk 3]
```diff
@@ -480,12 +480,7 @@ To add Mantle to your application:
...
  1. Drag and drop `Mantle.xcodeproj` into your application's Xcode project or
     workspace.
- 1. On the "Build Phases" tab of your application target, add Mantle to the
-    "Link Binary With Libraries" phase.
-    * **On iOS**, add `libMantle.a`.
-    * **On OS X**, add `Mantle.framework`. Mantle must also be added to any
-      "Copy Frameworks" build phase. If you don't already have one, simply add a
-      "Copy Files" build phase and target the "Frameworks" destination.
+ 1. On the "General" tab of your application target, add `Mantle.framework` to the "Embedded Binaries".
  1. Add `"$(BUILD_ROOT)/../IntermediateBuildFilesPath/UninstalledProducts/include" $(inherited)`
```
Comment (jspahrsummers) [inline README.md]: Sorry for missing it before, but I don't think this step or the next one are needed with frameworks. :tada:

File: README.md [hunk 4]
```diff
@@ -494,6 +489,11 @@ To add Mantle to your application:
     to add the appropriate Mantle target to the "Target Dependencies" of your
```
Comment (jspahrsummers) [inline README.md]: This step should also be unnecessary, because implicit dependency lookup works within one project too AFAIK.

**Comment by jspahrsummers** [comment]
Sorry, I found some more notes. :flushed: 

> If you add the Mantle xcodeproj to the top level of your workspace, you won't be able to drag/drop the Mantle.framework into the embedded binaries list.

You should be able to, from the other project's Products group. Or you can be explicit about it by clicking the + button inside Embedded Binaries, and selecting the framework manually.

**Comment by archagon** [comment]
@jspahrsummers I tried doing that. If I click + and select the framework manually in the Mantle project, it simply refuses to appear in the list. Dragging and dropping fails as well. Does it work for you?

**Comment by jspahrsummers** [comment]
@archagon I haven't tried on Xcode 6.3 completely from scratch, but that's how it's worked on every previous release. It sounds like an issue unique to your project, unfortunately.

Can you please share a link to your specific project, so we can investigate?

**Comment by archagon** [comment]
@jspahrsummers Unfortunately, it's a private project. I will try with a fresh project.

**Comment by archagon** [comment]
OK, here's the simplest possible test case. New workspace, new project, Mantle added as a submodule and subproject. Still can't add via + or drag/drop. https://github.com/archagon/mantle-installation-bug-demo

**Comment by dcaunt** [comment]
I've tried this from scratch with Xcode 6.3 and @archagon is correct, unfortunately it doesn't work. The only way that it works is to add Mantle.xcodeproj under the parent app's xcodeproj, rather than having the two as siblings.

Although Mantle.framework appears in the Add dialog, selecting it doesn't add it to the list. Drag and drop shows a green + on the cursor, but it fails here too.

File: README.md (hunk 3 above)
Comment (dcaunt) [inline README.md]: I think you're right, I checked and they don't appear to be necessary.

**Comment by jspahrsummers** [comment]
Sorry, I haven't had the time to go through the steps myself.

@dcaunt Does that mean we should rephrase the steps somehow? What's the concrete action to take here?

**Comment by dcaunt** [comment]
The simplest solution is to add `Mantle.xcodeproj` as a subproject to the main project. 

 Step 3 could be modified to explain this:

> Drag and drop Mantle.xcodeproj into your application's Xcode project or workspace. To work around an Xcode bug with embedded frameworks, add Mantle.xcodeproj as a subproject of your target project.

Facebook [describe the same issue](https://github.com/facebook/pop#ios-8-embedded-framework) in their pop framework. This [radar](http://www.openradar.appspot.com/19676555) appears related.

**Comment by jspahrsummers** [comment]
So should we just say:

> Drag and drop Mantle.xcodeproj into your application's Xcode project. Unfortunately, an [Xcode bug] means you should probably not add it to a workspace.

**Comment by dcaunt** [comment]
I think this is ready now :rocket:

**Comment by jspahrsummers** [comment]
:+1: Thanks for all the fixes!
